src/NuGet.Core/NuGet.Configuration/Settings/Items/AddItem.cs

```diff
@@ -76,6 +76,16 @@ public virtual string GetValueAsPath()
             return Value;
         }
 
+        public string GetConfigFilePath()
+        {
+            if (Origin != null)
+            {
+                return Origin.ConfigFilePath;
+            }
+
+            return string.Empty;
```

Comment:
From an API design point of view, why return an empty string?  If a setting was created programatically in-memory, so doesn't have a backing file, `null` feels like a reasonable return value. I think if I'm writing code, trying to get config file paths and I get an empty string that's more confusing.

But also, the method should have XMLDoc that explains this (it's really a shame we suppress the warnings telling us to always add xmldoc on public APIs)